runelite-client/src/main/java/net/runelite/client/plugins/thieving/ChestOverlay.java

```diff
@@ -0,0 +1,119 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.thieving;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ProgressPieComponent;
+
+@Singleton
+class ChestOverlay extends Overlay
+{
+	private final Client client;
+	private final ThievingPlugin plugin;
+	private final ThievingConfig config;
+
+	@Inject
+	private ChestOverlay(final Client client, final ThievingPlugin plugin, final ThievingConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		List<ChestRespawn> respawns = plugin.getRespawns();
+		if (respawns.isEmpty())
+		{
+			return null;
+		}
+
+		Instant now = Instant.now();
+		for (Iterator<ChestRespawn> it = respawns.iterator(); it.hasNext(); )
+		{
+			Color pieFillColor = config.respawnColor();
+			Color pieBorderColor;
+			ChestRespawn chestRespawn = it.next();
+
+			if (chestRespawn.getWorld() != client.getWorld())
+			{
+				continue;
+			}
+
+			float percent = (now.toEpochMilli() - chestRespawn.getStartTime().toEpochMilli()) / (float) chestRespawn.getRespawnTime();
+			if (percent > 1.0f)
+			{
+				it.remove();
+				continue;
+			}
+			WorldPoint worldPoint = chestRespawn.getWorldPoint();
+			LocalPoint loc = LocalPoint.fromWorld(client, worldPoint);
+			if (loc == null)
+			{
+				continue;
+			}
+
+			Point point = Perspective.localToCanvas(client, loc, client.getPlane(), 0);
+			if (point == null)
+			{
+				continue;
+			}
+
+			if (config.respawnPieInverted())
+			{
+				percent = 1.0f - percent;
+			}
+
+			pieBorderColor = pieFillColor.darker();
+
+			ProgressPieComponent ppc = new ProgressPieComponent();
+			ppc.setDiameter(config.respawnPieDiameter());
```

Comment:
if you set this on config changed rather than getting it every frame you won't need a config ref

Reply:
Updated